File: job-application-automation/src/cliQueue.ts
import { readFile } from 'node:fs/promises';
import { runQueue } from './queue.js';
import type { CandidateProfile } from './types.js';

/**
 * Usage:
 *   tsx src/cliQueue.ts <urls.txt> <profile.json> [--out ./out] [--captcha-timeout 300]
 *
 * urls.txt: one job URL per line (blank lines and lines starting with # are
 * ignored).
 *
 * Always runs headed — queue mode's whole point is a human reviewing and
 * clearing CAPTCHAs in one sitting, which requires a visible browser.
 */
async function main() {
  const [urlsPath, profilePath, ...rest] = process.argv.slice(2);

  if (!urlsPath || !profilePath) {
    console.error(
      'Usage: tsx src/cliQueue.ts <urls.txt> <profile.json> [--out ./out] [--captcha-timeout 300]',
    );
    process.exit(1);
  }

  const outIdx = rest.indexOf('--out');
  const outputDir = outIdx !== -1 ? rest[outIdx + 1] : './out';

  const timeoutIdx = rest.indexOf('--captcha-timeout');
  const captchaTimeoutSeconds =
    timeoutIdx !== -1 ? Number(rest[timeoutIdx + 1]) : undefined;

  const jobUrls = (await readFile(urlsPath, 'utf-8'))
    .split('\n')
    .map((line) => line.trim())
    .filter((line) => line && !line.startsWith('#'));

  const profile = JSON.parse(await readFile(profilePath, 'utf-8')) as CandidateProfile;

  const results = await runQueue({
    jobUrls,
    profile,
    outputDir,
    captchaTimeoutSeconds,
  });

  console.log('\n=== Summary ===');
  for (const r of results) {
    console.log(`${r.status.padEnd(16)} ${r.jobUrl}${r.error ? `  (${r.error})` : ''}`);
  }
}

main().catch((err) => {
  console.error(err instanceof Error ? err.message : err);
  process.exit(1);
});

File: job-application-automation-py/cli_queue.py
"""
Usage:
    python cli_queue.py <urls.txt> <profile.json> [--out ./out]
                         [--captcha-timeout 300] [--provider anthropic|groq]
                         [--model MODEL]

urls.txt: one job URL per line (blank lines and lines starting with # are
ignored).

Always runs headed -- queue mode's whole point is a human reviewing and
clearing CAPTCHAs in one sitting, which requires a visible browser.
"""

import argparse
import json
import sys

from queue_apply import run_queue
from types_ import CandidateProfile, QueueOptions


def main() -> None:
    parser = argparse.ArgumentParser()
    parser.add_argument("urls_path")
    parser.add_argument("profile_path")
    parser.add_argument("--out", default="./out")
    parser.add_argument("--captcha-timeout", type=int, default=300)
    parser.add_argument("--provider", choices=["anthropic", "groq"], default="anthropic")
    parser.add_argument("--model", default=None)
    args = parser.parse_args()

    with open(args.urls_path, encoding="utf-8") as f:
        job_urls = [
            line.strip()
            for line in f
            if line.strip() and not line.strip().startswith("#")
        ]

    with open(args.profile_path, encoding="utf-8") as f:
        profile = CandidateProfile.from_dict(json.load(f))

    results = run_queue(
        QueueOptions(
            job_urls=job_urls,
            profile=profile,
            output_dir=args.out,
            captcha_timeout_seconds=args.captcha_timeout,
            ai_provider=args.provider,
            ai_model=args.model,
        )
    )

    print("\n=== Summary ===")
    for r in results:
        suffix = f"  ({r.error})" if r.error else ""
        print(f"{r.status.ljust(16)} {r.job_url}{suffix}")


if __name__ == "__main__":
    try:
        main()
    except Exception as err:  # noqa: BLE001
        print(err, file=sys.stderr)
        sys.exit(1)
